pallets/file-system/src/types.rs

```diff
@@ -28,14 +27,34 @@ pub struct StorageRequestMetadata<T: Config> {
     /// Multiaddress of the user who requested the storage.
     ///
     /// SPs will expect a connection request to be initiated by the user with this multiaddress.
-    pub user_multiaddr: MultiAddress<T>,
-    /// List of BSPs that have volunteered to store the data.
-    pub bsps_volunteered: BoundedVec<StorageProviderId<T>, MaxBspsPerStorageRequest<T>>,
-    /// List of BSPs that have proven they are storing the data.
+    pub user_multiaddresses: BoundedVec<MultiAddress<T>, MaxMultiAddresses<T>>,
+    /// List of storage providers that can serve the data that is requested to be stored.
+    ///
+    /// This is useful when a BSP stops serving data and automatically creates a new storage request with no user multiaddresses, since
+    /// SPs can prove and serve the data to be replicated to other BSPs without the user having this stored on their local machine.
+    pub data_server_sps: BoundedVec<StorageProviderId<T>, MaxBspsPerStorageRequest<T>>, // TODO: Change the Maximum data servers to be the maximum SPs allowed
```

Comment:
They would know since when a storage request is generated they listen to it and can determine if there are `user_multiaddresses` or any `data_server_sps` that can serve the file. If there are none, and they have it, they can register themselves as a data server. 

If there aren’t any data servers, I like your idea to find where it is in the network of sps and “force” in some way to serve it I guess.